tracardi/service/storage/drivers/elastic/task.py

```diff
@@ -37,3 +37,6 @@ async def save_tasks(tasks):
 
 async def load_all(start: int = 0, limit: int = 100):
     return await storage_manager('task').load_all(start, limit)
+
+async def create(data: list):
+    return await storage_manager('task').upsert(data=data, replace_id=True)
```

Comment:
This is not done right. It should have all the logic. like this

```
return await storage_manager('task').upsert(data=[{
        "id": uuid4(),
        "timestamp": inserted_event.schedule.get_parsed_time(),
        "event": {
            "metadata": {
                "time": {
                    "insert": datetime.utcnow()
                }
            },
            "type": inserted_event.event.type,
            "properties": inserted_event.event.properties,
            "context": {"No context data provided.": None},
            "session": {
                "id": None
            },
            "source": {
                "id": inserted_event.source.id
            },
            "profile": {
                "id": None
            }
        },
        "status": None
    }], replace_id=True)
```
The driver should take all the required params to prepare a query and execute it. Please read About Single Responsibility principle. Outside driver there should be no logic referring saving in the database as we spoke before.  So the endpoint should not have the query dict as when we change the driver it will break the endpoint.